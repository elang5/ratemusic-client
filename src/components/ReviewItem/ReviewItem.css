.review {
  margin: 2rem 0;
  padding: 15px;
  border: 5px solid rgb(252, 60, 57);
  border-radius: 10px;
  display: flex;
  flex-flow: column wrap;
  align-items: center;
}

.review-title {
  text-decoration: underline;
}

.review > p, span {
  font-size: 18px;
}

.review-link {
  text-decoration: none;
  color: white;
  font-size: 20px;
}