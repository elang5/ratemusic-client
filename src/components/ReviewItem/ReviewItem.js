import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './ReviewItem.css'

export class ReviewItem extends Component {
  render() {
    const { review } = this.props
    return (
      <li className="review">
        <Link to={`/albums/${review.album_id}/reviews/${review.id}`} className="review-link">
          <p className="review-title">{review.title}</p>
        </Link>
          <span className="review-rating">{`Rating: ${review.rating}`}</span>
          <p className="user_info">{`By: ${review.user_name} on ${review.date_created}`}</p>
      </li>
    )
  }
}

export default ReviewItem
