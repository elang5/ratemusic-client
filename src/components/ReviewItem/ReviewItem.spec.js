import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReviewItem from './ReviewItem'

describe(`ReviewItem Component`, () => {

  it(`renders the ReviewItem component`, () => {
    const props = {
      review: {
        album_id: 1,
        title: 'Test',
        rating: 8,
        user_name: 'tester',
        date_created: '222'
      }
    }
    const wrapper = shallow(<ReviewItem {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})