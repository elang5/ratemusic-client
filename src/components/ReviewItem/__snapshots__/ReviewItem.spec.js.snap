// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItem Component renders the ReviewItem component 1`] = `
<li
  className="review"
>
  <Link
    className="review-link"
    to="/albums/1/reviews/undefined"
  >
    <h3
      className="review-title"
    >
      Test
    </h3>
  </Link>
  <AlbumStarRating
    rating={8}
  />
  <p
    className="user_info"
  >
    By: tester on 1/1/222
  </p>
</li>
`;
